src/ctapipe/image/muon/features.py

```diff
@@ -73,88 +107,243 @@ def intensity_ratio_inside_ring(
 
 
-def ring_containment(radius, center_x, center_y, camera_radius):
+def ring_containment(ring: MuonRingContainer, camera_radius: Quantity) -> float:
     """
-    Estimate angular containment of a ring inside the camera
-    (camera center is (0,0))
+    Estimate the angular containment of a muon ring inside the camera's field of view.
 
-    Improve: include the case of an arbitrary
-    center for the camera
-
-    See https://stackoverflow.com/questions/3349125/circle-circle-intersection-points
+    This function calculates the fraction of the muon ring that is contained within
+    the camera's field of view. It uses geometric properties to determine the intersection
+    of the ring with the circular boundary of the camera.
 
     Parameters
     ----------
-    radius: float or quantity
-        radius of the muon ring
-    center_x: float or quantity
-        x coordinate of the center of the muon ring
-    center_y: float or quantity
-        y coordinate of the center of the muon ring
-    camera_radius: float or quantity
-        radius of the camera
+    ring : MuonRingContainer
+        Container with the fitted ring parameters, including center coordinates and radius.
+    camera_radius : `astropy.units.Quantity`
+        Radius of the camera's field of view (in degrees).
 
     Returns
     -------
-    ringcontainment: float
-        the ratio of ring inside the camera
-    """
-    if hasattr(radius, "unit"):
-        radius, center_x, center_y, camera_radius = all_to_value(
-            radius, center_x, center_y, camera_radius, unit=radius.unit
-        )
-    d = np.sqrt(center_x**2 + center_y**2)
+    float
+        The fraction of the ring that is inside the camera's field of view, ranging from 0.0 to 1.0.
 
+    Notes
+    -----
+    The calculation is based on the geometric intersection of two circles:
+    the muon ring and the camera's field of view. The method handles cases where:
+    - The ring is fully contained within the camera.
+    - The ring is partially contained within the camera.
+    - The ring is completely outside the camera.
+
+    References
+    ----------
+    See https://stackoverflow.com/questions/3349125/circle-circle-intersection-points
+    for the geometric approach to circle-circle intersection.
+    """
     # one circle fully contained in the other
-    if d <= np.abs(camera_radius - radius):
+    if ring.center_distance <= np.abs(camera_radius - ring.radius):
         return 1.0
 
     # no intersection
-    if d > (radius + camera_radius):
+    if ring.center_distance > (ring.radius + camera_radius):
         return 0.0
 
-    a = (radius**2 - camera_radius**2 + d**2) / (2 * d)
-    return np.arccos(a / radius) / np.pi
+    a = (ring.radius**2 - camera_radius**2 + ring.center_distance**2) / (
+        2 * ring.center_distance
+    )
+    return np.arccos((a / ring.radius).to_value(u.dimensionless_unscaled)) / np.pi
+
+
+def ring_size_parameters(
```

Comment:
`ring_intensity_parameters`?

Reply:
I was thinking about this change, but this function also computes the number of pixels in the ring, so I left it as is. If you have a strong feeling that ring_intensity_parameters will be better, let me know and I will update.

Reply:
Yes, I think the `size` here came from when everything related to `intensity` was called `size`.

 So I'd say we rename